rest/endpoint/v1/api.go

```diff
@@ -0,0 +1,874 @@
+
+	keyspaceName := s.params(r, keyspaceParam)
+	tableName := s.params(r, tableParam)
+	columnName := s.params(r, "columnName")
+	user := auth.ContextUserOrRole(r.Context())
+
+	table, err := s.dbClient.DescribeTable(keyspaceName, tableName, user)
+	if err != nil {
+		msg := "unable to describe table"
+		s.logger.Debug(msg, "keyspace", keyspaceName, "table", tableName, "column", columnName, "error", err)
+
+		switch err.(type) {
+		case *e.NotFoundError:
+			RespondWithError(w, fmt.Errorf(msg), http.StatusNotFound)
+			return
+		default:
+			RespondWithError(w, fmt.Errorf(msg), http.StatusInternalServerError)
+			return
+		}
+	}
+
+	columns := columnMetadataToColumnDefinition(table.Columns)
+	var column m.ColumnDefinition
+	found := false
+	for _, col := range columns {
+		if col.Name == columnName {
+			column = col
+			found = true
+			break
+		}
+	}
+
+	if !found {
+		RespondWithError(w, fmt.Errorf("column '%s' not found in table", columnName), http.StatusNotFound)
+		return
+	}
+
+	RespondJSONObjectWithCode(w, http.StatusOK, column)
+}
+
+func (s *routeList) AddColumn(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+
+	keyspaceName := s.params(r, keyspaceParam)
+	tableName := s.params(r, tableParam)
+	user := auth.ContextUserOrRole(r.Context())
+
+	var columnDefinition m.ColumnDefinition
+	if err := parseAndValidatePayload(&columnDefinition, r); err != nil {
+		msg := "unable to parse payload"
+		s.logger.Debug(msg, "keyspace", keyspaceName, "table", tableName)
+		RespondWithError(w, errors.New(msg), http.StatusBadRequest)
+		return
+	}
+
+	column, err := m.ToDbColumn(columnDefinition)
+
+	if err != nil {
+		RespondWithError(w, err, http.StatusBadRequest)
+	}
+
+	tableInfo := db.AlterTableAddInfo{
+		Keyspace: keyspaceName,
+		Table:    tableName,
+		ToAdd:    []*gocql.ColumnMetadata{column},
+	}
+
+	err = s.dbClient.AlterTableAdd(&tableInfo, newDbOptions(user))
+	if err != nil {
+		msg := "unable to execute alter table query"
+		s.logger.Debug(msg, "keyspace", keyspaceName, "table", tableName, "error", err)
+
+		switch err.(type) {
+		case *e.ConflictError:
+			RespondWithError(w, fmt.Errorf(msg), http.StatusConflict)
+			return
+		default:
+			RespondWithError(w, fmt.Errorf(msg), http.StatusInternalServerError)
+			return
+		}
+	}
+
+	RespondJSONObjectWithCode(w, http.StatusCreated, m.TablesResponse{Success: true})
+}
+
+func (s *routeList) DeleteColumn(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+
+	keyspaceName := s.params(r, keyspaceParam)
+	tableName := s.params(r, tableParam)
+	columnName := s.params(r, "columnName")
+	user := auth.ContextUserOrRole(r.Context())
+
+	err := s.dbClient.AlterTableDrop(&db.AlterTableDropInfo{
+		Keyspace: keyspaceName,
+		Table:    tableName,
+		ToDrop:   []string{columnName},
+	}, newDbOptions(user))
+
+	if err != nil {
+		msg := "unable to execute alter table query"
```

Comment:
I'm wondering if this should return failures from the DB? This applies to other places too. An example where this makes sense: You cannot drop a column on a table with a materialized view. That's a useful error, but all that's returned is "unable to execute alter table query".

Reply:
Created: https://github.com/datastax/cassandra-data-apis/issues/129